### PR TITLE
Add query string-based filtering. Remove back buttons. Resolves #33.

### LINKED ISSUE #33: Back buttons don't work

### PR DESCRIPTION
**Pull request by tobyjsullivan**

### CODE REVIEW

File: src/angular/controllers/results.controller.js [hunk 1]
```diff
@@ -3,11 +3,28 @@ var controllers = angular.module('controllers');
...
-    // Filtered object based on the categories/regions 
-    $scope.results = Search.currResults()
+controllers.controller('ResultsCtrl', ['$scope', '$location', 'Search', function ($scope, $location, Search) {
     
+    // Filtered object based on the categories/regions
+    var getFilteredResults = function ($location) {
+        var filters = $location.search();
+
+        // Start by clearing all existing searches
+        Search.clearSelects();
```
Comment (lkysow) [inline src/angular/controllers/results.controller.js:13]: My latest merge has this done by default using `withClearAndEmit` in `Search`

File: src/angular/Services/search.js [hunk 2]
```diff
@@ -48,7 +48,18 @@ services.factory('Search', ['ServicesList', '$rootScope', function (ServicesList
             categoryDimension.filter(function(service) {
                 return service == category;
             });
-            $rootScope.$emit('FILTER_CHANGED')
+            $rootScope.$emit('FILTER_CHANGED');
+        },
+        selectRegion: function (region) {
+            regionDimension.filter(function (service) {
+                return service == region;
```
Comment (lkysow) [inline src/angular/Services/search.js:55]: So why were the other guys doing it using the geo stuff?

File: src/angular/Views/search-results.html [hunk 3]
```diff
@@ -1,11 +1,7 @@
 <!-- view when clicked once a category/region is click on -->
 
       <div data-ng-controller="ResultsCtrl">
-        <div class ="col-xs-12 col-sm-12"> 
-          <button id="backButton" type="button" class="btn btn-default" href="/">
-            <span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span> 
-            {{ 'BACK' | translate }}
-          </button>
```
Comment (lkysow) [inline src/angular/Views/search-results.html:8]: How do people go back to the search on mobile then?

**Comment by lkysow** [comment]
Stole your code and used it for https://github.com/PeaceGeeksSociety/services-advisor/pull/44 :) Thanks!

File: src/angular/Views/search-results.html (hunk 3 above)
Comment (tobyjsullivan) [inline src/angular/Views/search-results.html:8]: Mobile browsers have back buttons too

Comment (tobyjsullivan) [inline src/angular/Views/search-results.html:8]: Unless this is packaged into an app. Which would be new information for me.

**Pull request closed without merging**